plugins/out_kafka/kafka.c

```diff
@@ -178,7 +178,7 @@ int produce_message(struct flb_time *tm, msgpack_object *map,
                     /* Only if default topic is set and this topicname is not set for this message */
                     if (strncmp(topic->name, flb_kafka_topic_default(ctx)->name, val.via.str.size) == 0 &&
                         (strncmp(topic->name, val.via.str.ptr, val.via.str.size) != 0) ) {
-                        if (strstr(val.via.str.ptr, ",")) {
+                        if (flb_strnchr(val.via.str.ptr, ',', val.via.str.size)) {
```

Comment:
There is a C standard function `memchr(3)` that basically does what you want
to do here. So you may leverage it instead of reimplementing one by yourself.

```c
void *memchr(const void *s, int c, size_t n);

The  memchr()  function  scans  the  initial n bytes of the memory area
pointed to by s for the first instance of c.  Both c and the  bytes  of
the memory area pointed to by s are interpreted as unsigned char.
```